docs/js/main.js

```diff
@@ -144,112 +197,429 @@ function versionCompare(base, target) { // Return -1, 0, 1
     return -1
 }
 
+var searchBar;
+var searchIcon;
+
+// Load search link
+var linkParams = new URLSearchParams(window.location.href.replace("+", "%2B").split("?")[1]) // URLSearchParams decode '+' as space while encodeURI keeps + as is
+if (linkParams && linkParams.get("search")) {
+  setTimeout(() => {
+    let search = (linkParams.get("search"));
+    if (!search.includes("#")) { // hash links are higher priority
+      searchNow(search) // anchor link sometimes appear after the search param so filter it
+    }
+  }, 20) // Until searchBar is loaded
+}
+
 var content = document.getElementById("content");
 if (content) {
-  content.insertAdjacentHTML('afterbegin', '<input id="search-bar" type="text" placeholder="🔍 Search the documents.. (filters: v:1.0[.0][+])">');
+  let isNewPage = linkParams.get("isNew") != null;
+  content.insertAdjacentHTML('afterbegin', `<a id="search-icon" ${isNewPage ? 'class="search-icon-new"' : ""} title="Copy the search link."><img src="https://img.icons8.com/color/35/000000/search--v1.png"></a>`);
+  content.insertAdjacentHTML('afterbegin', `<span><input id="search-bar" ${isNewPage ? 'class="search-bar-version"' : ""} type="text" placeholder="Search the docs 🔍" title="Available Filters:&#13;&#10;&#13;&#10;Version:   v:2.5.3 v:2.2+ v:2.4-&#13;&#10;Type:      t:expression t:condition etc.&#13;&#10;New:       is:new"><span id="search-bar-after" style="display: none;">0 ${resultsFoundText}</span></span>`);
+  searchBar = document.getElementById("search-bar");
+  searchIcon = document.getElementById("search-icon");
+
+  if (isNewPage) {
+    let tags = []
+    let options = "<select id='search-version' name='versions' id='versions' onchange='checkVersionFilter()'></select>"
+    content.insertAdjacentHTML('afterbegin', `<span>${options}</span>`);
+    options = document.getElementById("search-version");
+    
+    let savedTags = getCookie("skVersions").split(",");
+    for (let i = 0; i < savedTags.length; i++) { // Append saved versions then check
+      if (savedTags[i] != "") {
+        let option = document.createElement('option')
+        option.value = savedTags[i]
+        option.textContent = "Since v" + savedTags[i]
+        options.appendChild(option)
+      }
+    }
+    if (savedTags && !linkParams.get("search") && !window.location.href.match(/.*?#.+/)) // Don't search for versions if the url has a search filter nor hash link
+      searchNow(`v:${savedTags[0]}+`) // Auto search on load
+    
+    setTimeout(() => {
+      $.getJSON("https://api.github.com/repos/SkriptLang/Skript/tags?per_page=83&page=2", (data) => { // 83 and page 2 matters to filter dev branches (temporary)
```

Comment:
On my machine it's still doing the request to GH each time I normally reload (/ go to tutorials and back) the page (although results were served from disk cache).